connectors/src/api/createConnector.ts

```diff
@@ -74,7 +75,33 @@ const _createConnectorAPIHandler = async (
       });
     }
 
-    return res.status(200).json({ connectorId: connectorRes.value });
+    const connector = await Connector.findByPk(connectorRes.value);
+    if (!connector) {
+      return apiError(req, res, {
+        api_error: {
+          type: "internal_server_error",
+          message: `Created connector not found in database. Connector id: ${connectorRes.value}`,
+        },
+        status_code: 500,
+      });
+    }
+    if (connector.type !== "notion") {
+      await reportInitialSyncProgress(parseInt(connectorRes.value), "0%");
```

Comment:
Would it make more sense to do this from your workflow instead ? I think ability to report status is specific to each provider, so it might make more sense to optionally initialize this when we set `lastSyncStartTime` for the first time on a connector

Reply:
Yes strongly bullish on invariant: no `if provider === X` in generic code :+1:

Reply:
👍

Reply:
This is not even needed anymore since we display `Synchronizing` with the `%` only if it is populated. I will just remove that line.